src/ZuluSCSI_disk.cpp

```diff
@@ -229,9 +229,22 @@ static void formatDriveInfoField(char *field, int fieldsize, bool align_right)
     }
 }
 
+// remove path and extension from filename 
+const char* get_image_name(const char* filename)
+{
+    char *j, *r;
+
+    r = (char *)malloc(strlen(filename));
```

Comment:
This seems to leak memory. I think it would be better to allocate a buffer for the result and pass a pointer to it as argument, similar to how e.g. `scsiDiskGetNextImageName()` works.